api/src/main/java/io/strimzi/api/kafka/model/JvmOptions.java

```diff
@@ -59,16 +58,6 @@ public void setXms(String xms) {
         this.xms = xms;
     }
 
-    @JsonProperty("-server")
-    @Description("-server option to to the JVM")
-    public Boolean isServer() {
```

Comment:
I don't think this change is backwards compatible in the sense that, IIRC, newer versions of kube will reject CRs which have properties which don't fit the schema. Someone using such a version of kube might have a CR with `-server: true` which was accepted with Strimzi `0.20.0` but rejected in `0.21.0`. Obviously we can remove all the code in the operator which consumes this value, and change the description. 

We were contemplating redesigning `JvmOptions` for `v1beta2` anyway, so I guess we should just go ahead and do that, adding `@PreventInVersions("v1alpha1-v1beta1")` to the property.

Reply:
@tombentley As I write in the description - this change completely disappeared from the API 2 years ago when the type was changed from `boolean` to `Boolean`. So it is not there for ~2 years and by that logic Kubernetes are rejecting the CRs with it since then. As you can see, this PR does not change any CRDs because it is long gone from them. Then I opened #4088 where I added it back. You suggested there that it should be removed instead if it is not needed anymore. So I closed #4088 and removed it in this PR. But if you want to instead have it added back, I can reopen #4088 again.

Reply:
Yes, sorry, you're quite right.